mono/utils/mono-threads-coop.c

```diff
@@ -67,10 +77,36 @@ mono_threads_state_poll (void)
 	}
 }
 
+static void *
+return_stack_ptr ()
+{
+	int i;
+	return &i;
+}
+
+static void
+copy_stack_data (MonoThreadInfo *info, void* stackdata_begin, void* stackdata_end)
```

Comment:
> Just add a local variable and use its address

That's what I had in my first iteration of the patch and Zoltan told me to call another function returning its local, `return_stack_ptr ` in the code. Shouldn't calling `return_stack_ptr ` be always safer since it will return a deeper stack pointer than the local variable?